packages/kbn-text-based-editor/src/text_based_languages_editor.tsx

```diff
@@ -143,6 +146,7 @@ let clickedOutside = false;
 let initialRender = true;
 let updateLinesFromModel = false;
 let lines = 1;
+let isDatePickerOpen = false;
```

Comment:
I would suggest making this and all the other `let` statements above it into React refs. Otherwise, you will run into problems if there is more than one instance of this component in the page because all instances of this component will share these module variables instead of having their own copies.

Reply:
Makes sense, but do you mind to do it in a tech dept PR? I am working on abig UI refactoring and doing this in this PR will cause huge conflicts on the other

Reply:
Sure, no problem

Reply:
Issue here https://github.com/elastic/kibana/issues/189029